numba_dpex/experimental/_kernel_dpcpp_spirv_overloads/_barriers_overloads.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-FileCopyrightText: 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Provides overloads for functions included in kernel_iface.barrier that
+generate dpcpp SPIR-V LLVM IR intrinsic function calls.
+"""
+from llvmlite import ir as llvmir
+from numba.core import cgutils, types
+from numba.extending import intrinsic, overload
+
+from numba_dpex.core import itanium_mangler as ext_itanium_mangler
+from numba_dpex.experimental.kernel_iface import (
+    group_barrier,
+    sub_group_barrier,
+)
+from numba_dpex.experimental.kernel_iface.memory_enums import (
+    MemoryOrder,
+    MemoryScope,
+)
+from numba_dpex.experimental.target import DPEX_KERNEL_EXP_TARGET_NAME
+
+from ._spv_atomic_inst_helper import get_memory_semantics_mask, get_scope
+
+
+def _get_memory_scope(fence_scope):
+    if isinstance(fence_scope, types.Literal):
+        return get_scope(fence_scope.literal_value)
+    return get_scope(fence_scope.value)
+
+
+@intrinsic
+def _intrinsic_barrier(
+    ty_context,  # pylint: disable=unused-argument
+    ty_exec_scope,  # pylint: disable=unused-argument
+    ty_mem_scope,  # pylint: disable=unused-argument
+    ty_spirv_mem_sem_mask,  # pylint: disable=unused-argument
+):
+    sig = types.void(types.uint32, types.uint32, types.uint32)
+
+    def _intrinsic_barrier_codegen(
+        context, builder, sig, args
+    ):  # pylint: disable=unused-argument
+        fn_name = "__spirv_ControlBarrier"
```

Comment:
Where is this function defined? Asking for learning purposes.

Reply:
It is defined in the sycl runtime.